cmd/state-transition-update/main.go

```diff
@@ -157,3 +170,31 @@ func runDefault() (rerr error) {
 
 	return nil
 }
+
+func addStateScript(cfg *config.Instance) error {
+	exec := appinfo.StateApp().Exec()
+	newInstallpath := cfg.GetString(installation.ConfigKeyInstallPath)
+
+	box := packr.NewBox("../../assets/state")
+	boxFile := "state.sh"
+	if runtime.GOOS == "windows" {
+		boxFile = "state.bat"
+		exec = strings.TrimSuffix(exec, exeutils.Extension) + ".bat"
+	}
+
+	tplParams := map[string]interface{}{
+		"path": filepath.Join(newInstallpath, filepath.Base(exec)),
+	}
+
+	fileBytes := box.Bytes(boxFile)
+	fileStr, err := strutils.ParseTemplate(string(fileBytes), tplParams)
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s template", boxFile)
+	}
+
+	if err = ioutil.WriteFile(exec, []byte(fileStr), 0755); err != nil {
```

Comment:
I think for Linux/Mac we'd also have to mark the file executable.

Reply:
Does the `0755` not cover this? The code here is very similar to the executor code [here](https://github.com/ActiveState/cli/blob/master/pkg/platform/runtime/executor/executor.go#L163)

Reply:
I suppose it should. If you tested it and it worked then I'd say we're good to go.

Reply:
I tested this code when it was in the installer and it worked on macOS. Once this is in place we should put it through and end to end test.